src/NServiceBus.RavenDB/Outbox/RavenDbOutboxStorage.cs

```diff
@@ -56,40 +56,42 @@ public OutboxCleaner(OutboxRecordsCleaner cleaner, TimeSpan frequencyToRunDedupl
                 this.timeToKeepDeduplicationData = timeToKeepDeduplicationData;
             }
 
-            protected override Task OnStart(IMessageSession session)
+            protected override Task OnStart(IMessageSession session, CancellationToken cancellationToken = default)
             {
                 if (frequencyToRunDeduplicationDataCleanup == Timeout.InfiniteTimeSpan)
                 {
                     return Task.CompletedTask;
                 }
 
-                cancellationTokenSource = new CancellationTokenSource();
-                cancellationToken = cancellationTokenSource.Token;
-
-                cleanupTask = Task.Run(() => PerformCleanup(), CancellationToken.None);
+                cleanupCancellationTokenSource = new CancellationTokenSource();
+                cleanupTask = Task.Run(() => PerformCleanup(cleanupCancellationTokenSource.Token), cancellationToken);
```

Comment:
Do we want this task to _not_ run if the token is cancelled? Note that the `OnStart` method will just return, without any exception, and the task will just not be running.